resilience4j-ratpack/src/main/java/io/github/resilience4j/ratpack/ratelimiter/RateLimiterConfigurationProperties.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright 2019 Dan Maas
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.github.resilience4j.ratpack.ratelimiter;
+
+import com.google.common.base.Strings;
+import io.github.resilience4j.core.ConfigurationNotFoundException;
+import io.github.resilience4j.core.lang.Nullable;
+import io.github.resilience4j.ratelimiter.RateLimiterConfig;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+
+public class RateLimiterConfigurationProperties {
```

Comment:
Can we share the code between Ratpack and Spring Boot and move it into resilience4j-framework-common?

Reply:
@RobWin there are a few framework differences regarding AOP ordering, BeanUtils, StringUtils, etc. The inner class `BackendProperties` appears to be similar as well, but in ratpack the setters use a builder pattern and in spring they don't. We can extract out the common stuff, but need to decide on 1 way to do things.

Reply:
For example:

```
        public BackendConfig baseConfig(String baseConfig) {
            this.baseConfig = baseConfig;
            return this;
        }
```

vs. 

```
		public void setBaseConfig(String baseConfig) {
			this.baseConfig = baseConfig;
		}
```

Reply:
Ok, I see. I think we should remove StringUtils, BeanUtils anyway. 
Micronaut does not have them by default as well.

Reply:
Can Ratpack use a Builder pattern with a `set` prefix?
Micronaut allows it.